migrations/versions/2025_03_05_2126-e4c05d7591a8_add_installation_table.py

```diff
@@ -0,0 +1,63 @@
+"""add installation table
+
+Revision ID: e4c05d7591a8
+Revises: 3ec2b4ab569c
+Create Date: 2025-03-05 21:26:19.034319+00:00
+
+"""
+
+from typing import Sequence, Union
+
+from alembic import op
+import sqlalchemy as sa
+
+
+# revision identifiers, used by Alembic.
+revision: str = "e4c05d7591a8"
+down_revision: Union[str, None] = "3ec2b4ab569c"
+branch_labels: Union[str, Sequence[str], None] = None
+depends_on: Union[str, Sequence[str], None] = None
+
+
+def upgrade() -> None:
+    op.execute("BEGIN TRANSACTION;")
+
+    op.execute(
+        """
+        CREATE TABLE IF NOT EXISTS instance (
+          id TEXT PRIMARY KEY,  -- UUID stored as TEXT
+          created_at DATETIME NOT NULL
+        );
+        """
+    )
+
+    op.execute(
+        """
+        -- The following trigger prevents multiple insertions in the
+        -- instance table. It is safe since the dimension of the table
+        -- is fixed.
+
+        CREATE TRIGGER single_instance
+        BEFORE INSERT ON instance
+        WHEN (SELECT COUNT(*) FROM instance) >= 1
+        BEGIN
+          SELECT RAISE(FAIL, 'only one instance!');
+        END;
+        """
+    )
```

Comment:
could we have a trigger that prevents deletion of the one row as well?

Reply:
I would avoid adding a constraint on deletion, it becomes cumbersome to modify the table since AFAIK you cannot disable triggers temporarily in sqlite, which means you'd have to rely on `DROP TABLE/CREATE TABLE` to fix any sort of issue.

Additionally, it's hard to delete that record by mistake since there's no `delete_instance` routine in `DbRecorder` and initialization takes care of recreating it.

I'm inclined to let this as is, but I'm not religious about it, we can add it now or later anyway.